installscripts/jazz-terraform-unix-noinstances/chefProvisioner.tf

```diff
@@ -13,76 +13,66 @@ resource "null_resource" "chef_provision_jenkins_server" {
-  #Because we have to provision a preexisting machine here and can't use the terraform ses command,
-  #we must use sed to insert AWS creds from the provisioner environment into a script chef will run later, before we copy the cookbook to the remote box.
+  # Update git branch and repo in jenkins cookbook
   provisioner "local-exec" {
-    command = "sed -i 's/AWS_ACCESS_KEY=.*.$/AWS_ACCESS_KEY='${var.aws_access_key}'/g' ${var.cookbooksSourceDir}/jenkins/files/credentials/aws.sh"
+    command = "sed -i 's|default\\['\\''git_branch'\\''\\].*.|default\\['\\''git_branch'\\''\\]='\\''${var.github_branch}'\\''|g' ${var.jenkinsattribsfile}"
   }
 
   provisioner "local-exec" {
-    command = "sed -i 's#AWS_SECRET_KEY=.*.$#AWS_SECRET_KEY='${var.aws_secret_key}'#g' ${var.cookbooksSourceDir}/jenkins/files/credentials/aws.sh"
+    command = "sed -i 's|default\\['\\''git_repo'\\''\\].*.|default\\['\\''git_repo'\\''\\]='\\''${var.github_repo}'\\''|g' ${var.jenkinsattribsfile}"
   }
 
-  # Update git branch and repo in jenkins cookbook
+  # Update AWS credentials in Jenkins Chef cookbook attributes
   provisioner "local-exec" {
-    command = "sed -i 's|default\\['git_branch'\\].*.|default\\['git_branch'\\]='${var.github_branch}'|g' ${var.jenkinsattribsfile}"
+    command = "sed -i 's|default\\['\\''aws_access_key'\\''\\].*.|default\\['\\''aws_access_key'\\''\\]='\\''${var.aws_access_key}'\\''|g' ${var.jenkinsattribsfile}"
   }
 
   provisioner "local-exec" {
-    command = "sed -i 's|default\\['git_repo'\\].*.|default\\['git_repo'\\]='${var.github_repo}'|g' ${var.jenkinsattribsfile}"
+    command = "sed -i 's|default\\['\\''aws_secret_key'\\''\\].*.|default\\['\\''aws_secret_key'\\''\\]='\\''${var.aws_secret_key}'\\''|g' ${var.jenkinsattribsfile}"
   }
 
-  # Update cognito script in cookbook
+  # Update cognito attribs in Jenkins Chef cookbook attributes
   provisioner "local-exec" {
-    command = "sed -i 's|<username>cognitouser</username>|<username>${var.cognito_pool_username}</username>|g' ${var.cookbooksSourceDir}/jenkins/files/credentials/cognitouser.sh"
+    command = "sed -i 's|default\\['\\''cognitouser'\\''\\].*.|default\\['\\''cognitouser'\\''\\]='\\''${var.cognito_pool_username}'\\''|g' ${var.jenkinsattribsfile}"
   }
 
   provisioner "local-exec" {
-    command = "sed -i 's|<password>cognitopasswd</password>|<password>${var.cognito_pool_password}</password>|g' ${var.cookbooksSourceDir}/jenkins/files/credentials/cognitouser.sh"
+    command = "sed -i 's|default\\['\\''cognitopassword'\\''\\].*.|default\\['\\''cognitopassword'\\''\\]='\\''${var.cognito_pool_password}'\\''|g' ${var.jenkinsattribsfile}"
   }
 
-  # Update sonar script in cookbook
+  #Update Gitlab attribs in Jenkins Chef cookbook attributes
   provisioner "local-exec" {
-    command = "sed -i 's|<username>sonaruser</username>|<username>${lookup(var.codeqmap, "sonar_username")}</username>|g' ${var.cookbooksSourceDir}/jenkins/files/credentials/sonar.sh"
```

Comment:
@bleggett We need to Update sonar script in cookbook also which is used for storing the sonarqube crenditals in the jenkins credentials.

Reply:
Right this got lost somehow, will pass them to chef via `default.rb` and then pass them to the script like with the others.

Reply:
Done.